core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.ts

```diff
@@ -285,36 +292,43 @@ angular.module('oppia').component('contributionsAndReview', {
         }
       };
 
-      var updateContributionSummaries = function() {
+      var getContributionSummaries = function(suggestionIdToSuggestions) {
         if (ctrl.activeSuggestionType === SUGGESTION_TYPE_TRANSLATE) {
-          ctrl.contributionSummaries = getTranslationContributionsSummary();
+          return getTranslationContributionsSummary(suggestionIdToSuggestions);
         } else if (ctrl.activeSuggestionType === SUGGESTION_TYPE_QUESTION) {
-          ctrl.contributionSummaries = getQuestionContributionsSummary();
+          return getQuestionContributionsSummary(suggestionIdToSuggestions);
         }
       };
 
       ctrl.switchToTab = function(tabType, suggestionType) {
+        ctrl.activeSuggestionType = suggestionType;
+        ctrl.activeTabType = tabType;
+        contributions = {};
+        ctrl.reloadOpportunitiesEventEmitter.emit();
+      };
+
+      ctrl.loadContributions = function() {
+        if (!ctrl.activeTabType || !ctrl.activeSuggestionType) {
```

Comment:
What case does this cover? Can you maybe extract the conditional to a small function with a descriptive name?

Reply:
This just blocks loading opportunities if the user isn't on any active tab.

Reply:
When can this occur? Just trying to understand when this function would be called when the user hasn't switched tabs yet.

Reply:
If someone calls the function before the component is initialized. This function will be called when the caller wants to load the opportunity.